data/base/endpoint/index/recursive.php

```diff
@@ -0,0 +1,24 @@
+<?php
+namespace Elementor\Data\Base\Endpoint\Index;
+
+use Elementor\Data\Base\Endpoint\Index;
+
+/**
+ * Class IndexRecursive, in cases where there no use of dynamic endpoints ( alpha/{id} )
+ * when the endpoints are static, eg:
+ * 'settings' - controller
+ * 'settings/products' - endpoint
+ * 'settings/partners' - endpoint
+ * Using 'IndexRecursive', the `get_items` method of index endpoint will run all over controller endpoints, except for himself,
+ * and call their `get_items` method.
+ * return example:
+ * [
+ *  'products' => [ product items ... ],
+ *  'partners' => [ partners list ... ],
+ * ]
+ */
+class Recursive extends Index {
+	public function get_items( $request ) {
+		return $this->controller->get_items_recursive( [ $this ] );
```

Comment:
MAybe move the `get_items_recursive` to here?

Reply:
Handled: https://github.com/elementor/elementor/pull/12820/commits/ba4d5d1d4ee775be0a15740691c8afc32b3f0058